glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/GleanLifecycleObserver.kt

```diff
@@ -4,17 +4,30 @@
 
 package mozilla.telemetry.glean.scheduler
 
+import android.content.Context
+import android.content.SharedPreferences
 import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.LifecycleEventObserver
 import androidx.lifecycle.LifecycleOwner
 import mozilla.telemetry.glean.Glean
 import mozilla.telemetry.glean.GleanMetrics.GleanBaseline
+import mozilla.telemetry.glean.GleanMetrics.GleanValidation
 
 /**
  * Connects process lifecycle events from Android to Glean's handleEvent
  * functionality (where the actual work of sending pings is done).
  */
-internal class GleanLifecycleObserver : LifecycleEventObserver {
+internal class GleanLifecycleObserver(
+    private val applicationContext: Context
+) : LifecycleEventObserver {
+    // Using SharedPreferences to track the status here should be less of an I/O hit
+    // compared to plain file usage, given that most of its operations are off the main
+    // thread. This also gives us the nice guarantee that once the function is called
+    // the OS will keep SharedPreferences working, even though the process is being killed.
+    internal val sharedPreferences: SharedPreferences by lazy {
+        applicationContext.getSharedPreferences("DirtyBitStatus", Context.MODE_PRIVATE)
```

Comment:
No exact idea how shared preferences work, should this have any "namespacing" using the "Glean" word?

Reply:
That's a good point. I did not want to use the class name, but having a Glean prefix sounds like a good idea :)